src/main/java/org/springframework/data/repository/reactive/RxJava2CrudRepository.java

```diff
@@ -42,98 +42,112 @@
+	 * @return {@link Flowable} emitting the saved entities.
+	 * @throws IllegalArgumentException in case the given {@link Iterable entities} or one of its entities is
+	 *           {@literal null}.
 	 */
 	<S extends T> Flowable<S> saveAll(Iterable<S> entities);
 
 	/**
 	 * Saves all given entities.
 	 *
 	 * @param entityStream must not be {@literal null}.
-	 * @return the saved entities.
-	 * @throws IllegalArgumentException in case the given {@code Publisher} is {@literal null}.
+	 * @return {@link Flowable} emitting the saved entities.
+	 * @throws IllegalArgumentException in case the given {@link Flowable entityStream} is {@literal null}.
 	 */
 	<S extends T> Flowable<S> saveAll(Flowable<S> entityStream);
 
 	/**
 	 * Retrieves an entity by its id.
 	 *
 	 * @param id must not be {@literal null}.
-	 * @return the entity with the given id or {@link Maybe#empty()} if none found.
-	 * @throws IllegalArgumentException if {@code id} is {@literal null}.
+	 * @return {@link Maybe} emitting the entity with the given id or {@link Maybe#empty()} if none found.
+	 * @throws IllegalArgumentException in case the given {@code id} is {@literal null}.
 	 */
 	Maybe<T> findById(ID id);
 
 	/**
 	 * Retrieves an entity by its id supplied by a {@link Single}.
 	 *
-	 * @param id must not be {@literal null}.
-	 * @return the entity with the given id or {@link Maybe#empty()} if none found.
-	 * @throws IllegalArgumentException if {@code id} is {@literal null}.
+	 * @param id must not be {@literal null}. Uses the first emitted element to perform the find-query.
+	 * @return {@link Maybe} emitting the entity with the given id or {@link Maybe#empty()} if none found.
+	 * @throws IllegalArgumentException in case the given {@link Single id} is {@literal null}.
 	 */
 	Maybe<T> findById(Single<ID> id);
 
 	/**
-	 * Returns whether an entity with the given id exists.
+	 * Returns whether an entity with the given {@code id} exists.
 	 *
 	 * @param id must not be {@literal null}.
-	 * @return {@literal true} if an entity with the given id exists, {@literal false} otherwise.
-	 * @throws IllegalArgumentException if {@code id} is {@literal null}.
+	 * @return {@link Single} emitting {@literal true} if an entity with the given id exists, {@literal false} otherwise.
+	 * @throws IllegalArgumentException in case the given {@code id} is {@literal null}.
 	 */
 	Single<Boolean> existsById(ID id);
 
 	/**
 	 * Returns whether an entity with the given id, supplied by a {@link Single}, exists.
 	 *
 	 * @param id must not be {@literal null}.
-	 * @return {@literal true} if an entity with the given id exists, {@literal false} otherwise.
-	 * @throws IllegalArgumentException if {@code id} is {@literal null}
+	 * @return {@link Single} emitting {@literal true} if an entity with the given id exists, {@literal false} otherwise.
+	 * @throws IllegalArgumentException in case the given {@link Single id} is {@literal null}.
```

Comment:
This should be either `{@literal IllegalArgument}` or `{@link IllegalArgument}`

Same in many other places.